docs/source/cpp/compute.rst

```diff
@@ -1231,6 +1231,28 @@ Containment tests
 * \(8) Output is true iff :member:`MatchSubstringOptions::pattern`
   matches the corresponding input element at any position.
 
+
+Hash Functions
+~~~~~~~~~~~~~~
+
+Not to be confused with the "group by" functions, Hash functions produce an array of hash
+values corresponding to the length of the input. Currently, these functions take a single
+array as input.
+
++---------------+-------+-------------+-------------+---------------+-------+
+| Function name | Arity | Input types | Output type | Options class | Notes |
++===============+=======+=============+=============+===============+=======+
+| hash32        | Unary | Any         | UInt32      |               | \(1)  |
++---------------+-------+-------------+-------------+---------------+-------+
+| hash64        | Unary | Any         | UInt64      |               | \(1)  |
```

Comment:
There are a few types (view, union, etc.) that are known to be unsupported, do we still want to mark supported input type as `Any` in the doc? This is kinda OK in the sense that we can think of these types are "temporarily unimplemented". But we can also explicitly reflect the actual supported types here. I'm not sure. Any thoughts @pitrou ?

Reply:
The supported index types should be documented in the "(1)" note IMHO.

Reply:
Going to update. Currently unsupported types are the view and union types. There are missing tests for temporal and decimal types.

Reply:
Maybe I am going to add support for sparse unions since that should be fairly straightforward.

Reply:
Run-end encoding also requires a custom hashing implementation.

Reply:
Updated the notes.